cpp/models/ode_secirvvs/parameters.h

```diff
@@ -644,82 +648,182 @@ class Parameters : public ParametersBase
+            }
+            if (this->get<ReducExposedImprovedImmunity>()[i] <= 0.0 ||
+                this->get<ReducExposedImprovedImmunity>()[i] > 1.0) {
+                log_warning("Constraint check: Parameter ReducExposedImprovedImmunity changed from {:0.4f} to {:d}",
+                            this->get<ReducExposedImprovedImmunity>()[i], 0);
+                this->get<ReducExposedImprovedImmunity>()[i] = 0;
+                corrected                                    = 1;
+            }
+            if (this->get<ReducInfectedSymptomsPartialImmunity>()[i] <= 0.0 ||
+                this->get<ReducInfectedSymptomsPartialImmunity>()[i] > 1.0) {
+                log_warning(
+                    "Constraint check: Parameter ReducInfectedSymptomsPartialImmunity changed from {:0.4f} to {:d}",
+                    this->get<ReducInfectedSymptomsPartialImmunity>()[i], 0);
+                this->get<ReducInfectedSymptomsPartialImmunity>()[i] = 0;
+                corrected                                            = 1;
+            }
+            if (this->get<ReducInfectedSymptomsImprovedImmunity>()[i] <= 0.0 ||
+                this->get<ReducInfectedSymptomsImprovedImmunity>()[i] > 1.0) {
+                log_warning(
+                    "Constraint check: Parameter ReducInfectedSymptomsImprovedImmunity changed from {:0.4f} to {:d}",
+                    this->get<ReducInfectedSymptomsImprovedImmunity>()[i], 0);
+                this->get<ReducInfectedSymptomsImprovedImmunity>()[i] = 0;
+                corrected                                             = 1;
+            }
+            if (this->get<ReducInfectedSevereCriticalDeadPartialImmunity>()[i] <= 0.0 ||
+                this->get<ReducInfectedSevereCriticalDeadPartialImmunity>()[i] > 1.0) {
+                log_warning("Constraint check: Parameter ReducInfectedSevereCriticalDeadPartialImmunity changed from "
+                            "{:0.4f} to {:d}",
+                            this->get<ReducInfectedSevereCriticalDeadPartialImmunity>()[i], 0);
+                this->get<ReducInfectedSevereCriticalDeadPartialImmunity>()[i] = 0;
+                corrected                                                      = 1;
+            }
+            if (this->get<ReducInfectedSevereCriticalDeadImprovedImmunity>()[i] <= 0.0 ||
+                this->get<ReducInfectedSevereCriticalDeadImprovedImmunity>()[i] > 1.0) {
+                log_warning("Constraint check: Parameter ReducInfectedSevereCriticalDeadImprovedImmunity changed from "
+                            "{:0.4f} to {:d}",
+                            this->get<ReducInfectedSevereCriticalDeadImprovedImmunity>()[i], 0);
+                this->get<ReducInfectedSevereCriticalDeadImprovedImmunity>()[i] = 0;
+                corrected                                                       = 1;
+            }
+            if (this->get<ReducTimeInfectedMild>()[i] <= 0.0 || this->get<ReducTimeInfectedMild>()[i] > 1.0) {
```

Comment:
Until here same question for all parameters: Shouldn't this be a <0?

Reply:
same as above